tests/test_default_start_date.py

```diff
@@ -0,0 +1,88 @@
+        Run a sync.
+        """
+        conn_id = connections.ensure_connection(self, original_properties=False)
+
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        streams_to_select = self.testable_streams(environment, data_type)
+
+        print("\n\nRUNNING {}".format(self.name()))
+        print("WITH STREAMS: {}".format(streams_to_select))
+        print("WITH START DATE: {}\n\n".format(self.START_DATE))
+
+        self.perform_and_verify_table_and_field_selection(
+            conn_id, found_catalogs, streams_to_select, select_all_fields=select_all_fields
+        )
+
+        return self.run_and_verify_sync(conn_id)
+
+    def test_run(self):
+        """
+        Verify that for each stream you can call sync just so it exercises the code.
+        """
+        default_start_date = dt.utcnow() - timedelta(days=365) # 1 year ago
+
+        # Testing Default start date (1 year ago)
+        self.START_DATE = dt.strftime(default_start_date, self.START_DATE_FORMAT)
+        self.set_environment(self.SANDBOX)
+        self.default_start_date_test(DataType.DYNAMIC, self.testable_streams_dynamic().difference(self.production_streams()))
+        self.default_start_date_test(DataType.STATIC, self.testable_streams_static().difference(self.production_streams()))
+        self.set_environment(self.PRODUCTION)
+        self.default_start_date_test(DataType.DYNAMIC, self.testable_streams_dynamic().difference(self.sandbox_streams()))
+        self.default_start_date_test(DataType.STATIC, self.testable_streams_static().difference(self.sandbox_streams()))
+
+        days_prior = random.randint(7, 358) # 1 week to 57 weeks
+
+        # Testing start date prior to default
+        self.START_DATE = dt.strftime(default_start_date - timedelta(days=days_prior), self.START_DATE_FORMAT)
+        self.set_environment(self.SANDBOX)
+        self.default_start_date_test(DataType.DYNAMIC, self.testable_streams_dynamic().difference(self.production_streams()))
+        self.default_start_date_test(DataType.STATIC, self.testable_streams_static().difference(self.production_streams()))
+        self.set_environment(self.PRODUCTION)
+        self.default_start_date_test(DataType.DYNAMIC, self.testable_streams_dynamic().difference(self.sandbox_streams()))
+        self.default_start_date_test(DataType.STATIC, self.testable_streams_static().difference(self.sandbox_streams()))
+
+    def default_start_date_test(self, data_type, testable_streams):
+        streams_without_data = self.untestable_streams()
+
+        # Verify we can run a check and sync with the given start date
+        record_count_by_stream = self.run_standard_sync(self.get_environment(), data_type)
+
+        # Verify we get records for streams with data
+        for stream in testable_streams:
+            if stream in streams_without_data:
+                self.assertEqual(0, record_count_by_stream.get(stream, 0))
```

Comment:
Can we do greater than or equal to? No reason to fail if records magically show up...

Reply:
Not sure I agree, if records did show up for those 3 untestable streams I think we'd want to know.

Reply:
Why? Why would we want this test to fail for that reason?

Reply:
Well generally, I think it is better to be as strict as we can and ease back only when necessary like when we hit instability like we are seeing elsewhere now. I think in this case it would be bad if we randomly get data for those untestable streams. Our expectations would be wrong, we are skipping all the feature tests for these streams due to the fact that we can't get data into the account for them. If something drastic enough changes that data appears, I think that should require someone's attention. Either square is messing up in a very bad way, or our expectations are bad and we shouldn't be skipping all those tests.

Reply:
I don't understand why we want daily builds to fail for this reason. That would mean the test would probably be failing for months until someone investigates. And without context it's hard to determine exactly why it's failing quickly. If there are multiple things that might have started failing between that failure and by the time someone looks at it it becomes harder to distinguish test failure reasons...

Reply:
I made the change here but I don't totally agree with it. I think I understand your point about the test maintenance issue, but I am not sure if writing assertions that break our expectations is good path to address it.

Reply:
Agree to disagree 🤷